core/src/main/java/com/linecorp/armeria/client/ClientFactoryBuilder.java

```diff
@@ -765,6 +776,33 @@ private ClientFactoryOptions buildOptions() {
         }
     }
 
+    private static Map<ChannelOption<?>, Object> getDefaultNettyChannelOptions(
+            ClientFactoryOptions clientFactoryOptions, long maxConnectionAgeMillis,
+            long idleTimeoutMillis, long pingIntervalMillis) {
+        final Map<ChannelOption<?>, Object> userDefinedChannelOptions = clientFactoryOptions.channelOptions();
+        final Builder<ChannelOption<?>, Object> newChannelOptionsBuilder = ImmutableMap.builder();
+
+        if (Flags.transportType() == TransportType.EPOLL) {
```

Comment:
Gave this a try - I'm yet to test how this will work when `netty-incubator-transport-native-io_uring` is excluded though.
(in particular, I'm thinking of #3244)

Reply:
> I'm yet to test how this will work when netty-incubator-transport-native-io_uring is excluded though.

`Flags.transportType()` can be `IO_URING` only when `netty-incubator-transport-native-io_uring` is available. https://github.com/line/armeria/blob/master/core/src/main/java/com/linecorp/armeria/common/Flags.java#L437
So can't we just ignore such a situation?

Reply:
Could add a separate module under `:it`? But I guess it'll be tested only on a machine with a very modern Linux kernel, which is not the case for our CI machines. So, how about just adding `|| ... == IO_URING` and wait until we get a bug report?